AppController/djinn.rb

```diff
@@ -5785,7 +5797,13 @@ def get_node_stats_json(secret)
 
     # Get stats from SystemManager.
     imc = InfrastructureManagerClient.new(secret)
-    system_stats = JSON.load(imc.get_system_stats)
+    begin
+      system_stats = JSON.load(imc.get_system_stats)
+    rescue SOAP::FaultError, FailedNodeException => exception
```

Comment:
If we could recognize BadSecret error it would be nice to restart InfrastructureManager. Sometimes the manager is not restarted properly and continues to use old secret.